mathesar/api/serializers/constraints.py

```diff
@@ -129,3 +144,13 @@ def get_mapping_field(self, data):
         else:
             constraint_type = data.get('type', None)
         return constraint_type
+
+    def run_validation(self, data):
+        columns = data.get('columns', None)
+        if columns == []:
+            message = 'Columns field cannot be empty'
+            raise MathesarValidationException(
+                ValidationError,
+                message=message
+            )
```

Comment:
@silentninja Also right here I am getting an error: 'ReturnList' object has no attribute 'get'
Is this occurring due to the constraint viewset?

Reply:
I am not able to reproduce this bug. I tried sending a constraints request with an empty column field and got the correct error response. Can you write some test cases to highlight the bugs

Reply:
@silentninja The error seems to occur only while sending the request through the DRF API client. The response in the test seems correct.

Reply:
@silentninja The error seems to occur only while sending the request through the DRF API client. The response in the test seems correct.

Reply:
@Anish9901 Can you see if the frontend is facing that issue, it could be due to the content type of the request.

Reply:
Yes, it's a frontend-specific issue, I'll create an issue for it.